docs/index.html

```diff
@@ -31,19 +31,76 @@
   <meta name="msapplication-TileColor" content="#da532c">
   <meta name="msapplication-TileImage" content="./logos/mstile-144x144.png">
   <meta name="theme-color" content="#ffffff">
-  <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/devicons/devicon/devicon.min.css">
   <link rel="stylesheet" href="./assets/css/style.css">
   <meta property="og:title" content="Devicon" />
   <meta property="og:image" content="https://raw.githubusercontent.com/devicons/devicon/master/icons/devicon/devicon-original-wordmark.svg" />
   <meta property="og:url" content="https://devicon.dev" />
   <meta property="og:type" content="website" />
   <meta property="og:locale" content="en_US" />
   <meta property="og:description" content="Devicon aims to gather all logos representing development languages and tools in one font. 🚀" />
+  <script>
+    /////////////////////////////
+    // Basic helper functions //
+    /////////////////////////////
+    function addCss(href) {
+      var link = document.createElement('link');
+
+      link.rel = 'stylesheet';
+      link.type = 'text/css';
+      link.href = `${href}`;
+
+      document.body.appendChild(link);
+      document.head.appendChild(link);
+    }
+
+    function addImage(src, id, img_class) {
+      var img = document.createElement("img");
+
+      img.src = src;
+      img.alt = "{{selectedIcon.name}}-{{svgVersion}}";
+      img.classList.add(img_class);
+
+      document.getElementById(id).appendChild(img);
+    }
+
+    // Unused. May be useful in the future.
+    function addElement(content, id) {
+      document.getElementById(id).innerHTML = content;
+    }
+
+    ///////////////////////
+    // Path management //
+    /////////////////////
+    var path = '/devicon.min.css';
+    var baseUrl
+  
+    var xhr = new XMLHttpRequest();
+    xhr.open('HEAD', path, false);
+    xhr.send();
+
+    // Make sure the file exists, otherwise, use latest CDN link.
+    if (xhr.status == "404") {
+      baseUrl = `https://cdn.jsdelivr.net/gh/devicons/devicon@latest/`;
+    } else {
+      baseUrl = `/`;
+    }
+
+    // Temporary solution.
+    var counter = 0
+    var timer = setInterval(function() {
+          ++counter;
+
+          addCss(`${baseUrl}devicon.min.css`);
+          addImage(`${baseUrl}icons/{{selectedIcon.name}}/{{selectedIcon.name}}-{{svgVersion}}.svg`, "svg-list", 'svgLogo');
+
+          if (counter == 1) {
+            clearInterval(timer);
+          }
+        }, 0);
+  </script>
 </head>
 
 <body ng-app="devicon" ng-controller="IconListCtrl">
-  <link rel="stylesheet" ng-if="latestReleaseTagging" ng-href="https://cdn.jsdelivr.net/gh/devicons/devicon@{{ latestReleaseTagging }}/devicon.min.css">
```

Comment:
Is this not needed anymore? Could you explain what it did, and why we don't need it anymore?

Reply:
1. Yes it is needed, but it is added by the script above.
2. The stylesheet file (`devicon.min.css`) loads all the icons data, colors, etc..
3. As I said, it is added by the JavaScript script above.

Reply:
I see. It would probably be less code to move the script to the .js file and add a variable, then use that variable here instead of adding the element via the DOM.
That would more closely follow the SOLID principle too

Reply:
So, you're saying to move the whole script to another JS file and load the script via HTML?

Reply:
Yes, sort of. Moving it to the .js file, but using Angular features instead of interacting directly with the DOM. It's less complex and less code that way.